storage/src/interface.rs

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! High-level application-agnostic storage interface
+
+use crate::schema::{self, Schema};
+use storage_core::{backend, Backend, Data, DbIndex};
+
+/// The main storage type
+pub struct Storage<B: Backend, Sch> {
+    backend: B::Impl,
+    _schema: core::marker::PhantomData<Sch>,
```

Comment:
I'm just curious: is an underscore prefix really required for a `PhantomData` field?

Reply:
I don't think it's strictly required. It's there to suggest the field is not really used conceptually.